Code/GraphMol/AddHs.cpp

```diff
@@ -319,7 +319,8 @@ void setHydrogenCoords(ROMol *mol, unsigned int hydIdx, unsigned int heavyIdx) {
 }  // end of unnamed namespace
 
 namespace MolOps {
-void addHs(RWMol &mol, bool explicitOnly, bool addCoords) {
+void addHs(RWMol &mol, bool explicitOnly, bool addCoords,
+           const UINT_VECT *onlyOnAtoms) {
```

Comment:
You're right.  We could always make wrapper functions to call the one you have here, so I have no problem with the current layout going forwards.  I do find though that the new parameters I add at the end of a function usually become up being the ones that should have gone first ;)

----
Brian Kelley

> On Feb 11, 2016, at 10:51 PM, Greg Landrum <notifications@github.com> wrote:
> 
> In Code/GraphMol/AddHs.cpp:
> 
> > @@ -319,7 +319,8 @@ void setHydrogenCoords(ROMol *mol, unsigned int hydIdx, unsigned int heavyIdx) {
> >  }  // end of unnamed namespace
> >  
> >  namespace MolOps {
> > -void addHs(RWMol &mol, bool explicitOnly, bool addCoords) {
> > +void addHs(RWMol &mol, bool explicitOnly, bool addCoords,
> > +           const UINT_VECT *onlyOnAtoms) {
> Rather than starting down this road here with a single function (actually a couple of functions), I'd rather have a longer design discussion about better ways to deal with functions/methods that take a lot of optional arguments.
> 
> But this functionality doesn't need to be gated on that discussion taking place.
> 
> —
> Reply to this email directly or view it on GitHub.
>